src/Microsoft.AspNet.Diagnostics/DeveloperExceptionPage/DeveloperExceptionPageExtensions.cs

```diff
@@ -24,23 +23,31 @@ public static IApplicationBuilder UseDeveloperExceptionPage(this IApplicationBui
                 throw new ArgumentNullException(nameof(builder));
             }
 
-            return builder.UseDeveloperExceptionPage(new ErrorPageOptions());
+            return builder.UseDeveloperExceptionPage(options => { });
         }
 
         /// <summary>
-        /// Captures synchronous and asynchronous exceptions from the pipeline and generates HTML error responses.
-        /// Full error details are only displayed by default if 'host.AppMode' is set to 'development' in the IApplicationBuilder.Properties.
+        /// Captures synchronous and asynchronous <see cref="Exception"/> instances from the pipeline and generates HTML error responses.
         /// </summary>
-        /// <param name="builder"></param>
-        /// <param name="options"></param>
-        /// <returns></returns>
-        public static IApplicationBuilder UseDeveloperExceptionPage(this IApplicationBuilder builder, ErrorPageOptions options)
+        /// <param name="builder">The <see cref="IApplicationBuilder"/>.</param>
+        /// <param name="configureOptions">A callback to configure <see cref="ErrorPageOptions"/>.</param>
+        /// <returns>A reference to the <paramref name="builder"/> after the operation has completed.</returns>
+        public static IApplicationBuilder UseDeveloperExceptionPage(
+            this IApplicationBuilder builder,
+            Action<ErrorPageOptions> configureOptions)
         {
             if (builder == null)
             {
                 throw new ArgumentNullException(nameof(builder));
             }
 
+            if (configureOptions == null)
+            {
+                throw new ArgumentNullException(nameof(configureOptions));
+            }
+
+            var options = new ErrorPageOptions();
```

Comment:
Should we rename these? Using this overload is already broken for end users, so we might make the most of it and change the type name to match the middleware name. Thoughts?